demo/meeting/src/providers/NavigationProvider.tsx

```diff
@@ -38,6 +42,20 @@ const NavigationProvider = ({ children }: Props) => {
   const [showMetrics, setShowMetrics] = useState(false);
   const isDesktopView = useRef(isDesktop());
 
+  const history = useHistory();
+  const meetingManager = useMeetingManager();
+  useEffect(() => {
+    const handleBackOrForwardClick = (e: any) => {
+      history.push(routes.HOME);
+      meetingManager.leave();
+    }
+    window.onpopstate = (e: any) => handleBackOrForwardClick(e);
+  }, []);
+
+  if (!window.location.href.includes(routes.MEETING || routes.DEVICE)) {
+    meetingManager.leave()
+  };
```

Comment:
I think we can make this a little simpler if we use the `useLocation` hook. Really we just want to do the following whenever the location changes
- check if we're on the home page
- decide whether or not to clear the meeting

Reply:
I ran into a couple of problems when trying that way originally. Clearing the meeting in `AppState` doesn't actually cause the user to leave a meeting. Even when the meeting is cleared, the user can still be seen and/or heard in the meeting. We need to call `leave()` from the MeetingManager as far as I can tell. 

Also, when a user presses the back button in a meeting, they don't go to the home screen, they go to the device setup screen. We can't simple see if they are in this screen and clear the meeting, because on the way into a meeting they visit this screen and need the meeting data. Does that make sense?

Reply:
Right, I meant end/clear the meeting session (calling leave or w/e the API is on MeetingManager)

Yeah, that makes sense. I think you can still achieve that with the `useLocation` hook

```
useEffect(() => {
  if (location.pathName === "MEETING" or w/e) {
  return () => { 
  // Our route has changed from meeting to something else, 
  // so we can do our logic here like stopping the meeting
}
}, [location])
```

Or we could just not stop the meeting on the device screen, since I think technically it "should" work even if you click back since it has to do with selecting devices and quality and such. Mainly I think it should be cleared on the home screen

Reply:
I'm assuming in your snippet above it' supposed to be `if (location.pathName !== "MEETING" or w/e) {`???

I can switch it to use useLocation for sure, but I don't think the user clicking back, going to the device setup screen, and being in the meeting (including sending video and audio) is a good experience. You shouldn't ever send media to a meeting from that screen. I think we do need to hook up the back button to leaving the meeting.

Reply:
No, the snippet I posted was intentionally `=== "MEETING"

We're returning a cleanup function we want to run when our state has changed FROM meeting to anything else